### ISSUE TITLE
Snapshot: UITest ok on Xcode but failed on snapshot

### ISSUE THREAD

**Issue by GUL-** [fastlane/fastlane#4708]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
I was using successfully snapshot to take screenshots of all device sizes for both en and fr locale.
I have a single UI test case `testTakeScreenshots` which basically go through the app and take 4 screenshots.
This tests pass successfully when running on Xcode but this fails running with snapshot and I cannot figure out why.

##### Complete output when running fastlane, including the stack trace and command used
```shell
testTakeScreenshots, UI Testing Failure - Failure getting snapshot Error Domain=XCTestManagerErrorDomain Code=9 "Error -25204 getting snapshot for element <AXUIElement 0x7fca0ae146f0> {pid=28169}" UserInfo={NSLocalizedDescription=Error -25204 getting snapshot for element <AXUIElement 0x7fca0ae146f0> {pid=28169}}
<unknown>:0
```
```


	 Executed 1 test, with 1 failure (0 unexpected) in 6.084 (6.087) seconds

ERROR [2016-05-17 21:52:09.52]: Exit status: 65
ERROR [2016-05-17 21:52:09.53]: Tests failed - check out the log above
ERROR [2016-05-17 21:52:09.53]: Backtrace:
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/fastlane_core-0.41.3/lib/fastlane_core/ui/interface.rb:139:in `user_error!'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/fastlane_core-0.41.3/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/error_handler.rb:10:in `handle_test_error'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:172:in `block in launch'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/fastlane_core-0.41.3/lib/fastlane_core/command_executor.rb:85:in `call'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/fastlane_core-0.41.3/lib/fastlane_core/command_executor.rb:85:in `execute'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:166:in `launch'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:66:in `run_for_device_and_language'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:45:in `block (3 levels) in work'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:37:in `each'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:37:in `block (2 levels) in work'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:36:in `each'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:36:in `block in work'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:35:in `each'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:35:in `work'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/bin/snapshot:36:in `block (2 levels) in run'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/fastlane_core-0.41.3/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/bin/snapshot:96:in `run'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/bin/snapshot:110:in `<top (required)>'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/bin/snapshot:23:in `load'
	/Users/GUL/.rvm/rubies/ruby-2.1.3/bin/snapshot:23:in `<main>'
	/Users/GUL/.rvm/gems/ruby-2.1.3/bin/ruby_executable_hooks:15:in `eval'
	/Users/GUL/.rvm/gems/ruby-2.1.3/bin/ruby_executable_hooks:15:in `<main>'

+----------+---------+
|  snapshot results  |
+----------+---------+
| Device   | fr-FR   |
+----------+---------+
| iPhone 6 |  ❌      |
+----------+---------+

Successfully submitted a crash report. If this is a problem with one of the tools specifically,
please submit an issue on GitHub and attach the following number to it: 'fd9b91c7c71867c72260d25d90fbb7a7'
The crash report has been stored locally '/tmp/sentry_fd9b91c7c71867c72260d25d90fbb7a7.json'
/Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:52:in `work': [!] Tests failed - check out the log above (RuntimeError)
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/bin/snapshot:36:in `block (2 levels) in run'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/fastlane_core-0.41.3/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/bin/snapshot:96:in `run'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/lib/ruby/gems/2.1.0/gems/snapshot-1.12.2/bin/snapshot:110:in `<top (required)>'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/bin/snapshot:23:in `load'
	from /Users/GUL/.rvm/rubies/ruby-2.1.3/bin/snapshot:23:in `<main>'
	from /Users/GUL/.rvm/gems/ruby-2.1.3/bin/ruby_executable_hooks:15:in `eval'
	from /Users/GUL/.rvm/gems/ruby-2.1.3/bin/ruby_executable_hooks:15:in `<main>'
```

### Configuration Files

Snapfile:
```ruby
# A list of devices you want to take the screenshots from
 devices([
   "iPhone 6",
   #"iPhone 6 Plus",
   #"iPhone 5",
   #"iPhone 4s",
 ])

languages([
  #"en-US",
  "fr-FR"
])

# Arguments to pass to the app on launch. See https://github.com/fastlane/snapshot#launch-arguments
# launch_arguments(["-favColor red"])

# The name of the scheme which contains the UI Tests
scheme "usiappUITests"

# Where should the resulting screenshots be stored?
output_directory "./fastlane/screenshots"

clear_previous_screenshots true # remove the '#' to clear all previously generated screenshots before creating new ones

# Choose which project/workspace to use
# project "./Project.xcodeproj"
workspace "./usiapp.xcworkspace"

# For more information about all available options run
# snapshot --help
```

### Environment

fastlane version (run `fastlane -v`):
fastlane 1.89.0

snapshot version 
snapshot 1.12.2

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?
no

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?
no

**Comment by wolffan** [comment]
Got similar issue.
Tests work on xcode but not when launched from CI

**Comment by vpolouchkine** [comment]
> I was using successfully snapshot to take screenshots of all device sizes for both en and fr locale

Did you update `snapshot`, eg. it was working before?

**Comment by esetnik** [comment]
I'm having a similar issue. Each of my `snapshot` retries fails with:
```
** TEST FAILED **
[17:40:50]: Exit status: 65
[17:40:50]: Tests failed - check out the log above
```

This is my first time attempting `snapshot` though

**Comment by esetnik** [comment]
```** TEST FAILED **
ERROR [2016-05-18 18:06:49.42]: Exit status: 65
ERROR [2016-05-18 18:06:49.42]: Tests failed - check out the log above
ERROR [2016-05-18 18:06:49.42]: Backtrace:
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/fastlane_core-0.43.5/lib/fastlane_core/ui/interface.rb:139:in `user_error!'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/fastlane_core-0.43.5/lib/fastlane_core/ui/ui.rb:14:in `method_missing'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/error_handler.rb:10:in `handle_test_error'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:172:in `block in launch'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/fastlane_core-0.43.5/lib/fastlane_core/command_executor.rb:88:in `call'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/fastlane_core-0.43.5/lib/fastlane_core/command_executor.rb:88:in `execute'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:166:in `launch'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:66:in `run_for_device_and_language'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:72:in `rescue in run_for_device_and_language'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:66:in `run_for_device_and_language'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:45:in `block (3 levels) in work'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:37:in `each'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:37:in `block (2 levels) in work'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:36:in `each'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:36:in `block in work'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:35:in `each'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:35:in `work'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/bin/snapshot:36:in `block (2 levels) in run'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/fastlane_core-0.43.5/lib/fastlane_core/ui/fastlane_runner.rb:26:in `run!'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/bin/snapshot:96:in `run'
	/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/bin/snapshot:110:in `<top (required)>'
	/usr/local/var/rbenv/versions/2.2.3/bin/snapshot:22:in `load'
	/usr/local/var/rbenv/versions/2.2.3/bin/snapshot:22:in `<main>'

+-----------+---------+
|  snapshot results   |
+-----------+---------+
| Device    | en-US   |
+-----------+---------+
| iPhone 6s |  ❌      |
+-----------+---------+

Successfully submitted a crash report. If this is a problem with one of the tools specifically,
please submit an issue on GitHub and attach the following number to it: '4fd3df222451eb3de5a9c6f02dcece48'
The crash report has been stored locally '/tmp/sentry_4fd3df222451eb3de5a9c6f02dcece48.json'
/usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/lib/snapshot/runner.rb:52:in `work': [!] Tests failed - check out the log above (RuntimeError)
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/bin/snapshot:36:in `block (2 levels) in run'
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/fastlane_core-0.43.5/lib/fastlane_core/ui/fastlane_runner.rb:26:in `run!'
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/bin/snapshot:96:in `run'
	from /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/snapshot-1.12.2/bin/snapshot:110:in `<top (required)>'
	from /usr/local/var/rbenv/versions/2.2.3/bin/snapshot:22:in `load'
	from /usr/local/var/rbenv/versions/2.2.3/bin/snapshot:22:in `<main>'```

**Comment by TKBurner** [comment]
@esetnik Thanks for sharing can you provide the full output including commands you entered and were entered automatically by Fastlane? Thanks!

**Comment by esetnik** [comment]
Hi @TKBurner sorry for the delay. Is this what you're looking for?

```
✗ snapshot
[19:43:18]: xcrun xcodebuild -list -workspace './project.xcworkspace'

+----------------------------+------------------------------------------------------------------+
|                                  Summary for snapshot 1.12.2                                  |
+----------------------------+------------------------------------------------------------------+
| devices                    | ["iPhone 6s"]                                                    |
| languages                  | ["en-US"]                                                        |
| scheme                     | projectUITests                                                |
| clear_previous_screenshots | true                                                             |
| clean                      | true                                                             |
| workspace                  | ./project.xcworkspace                                         |
| output_directory           | /Users/dev/work/project/project_ios/fastlane/screenshots |
| launch_arguments           | [""]                                                             |
| ios_version                | 9.3                                                              |
| skip_open_summary          | false                                                            |
| reinstall_app              | false                                                            |
| erase_simulator            | false                                                            |
| localize_simulator         | false                                                            |
| app_identifier             | com.project.project                                        |
| buildlog_path              | ~/Library/Logs/snapshot                                          |
| number_of_retries          | 1                                                                |
| stop_after_first_error     | false                                                            |
+----------------------------+------------------------------------------------------------------+

[19:43:19]: Clearing previously generated screenshots
[19:43:19]: Building and running project - this might take some time...
[19:43:20]: Patching '/Users/dev/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
[19:43:23]: -------------------------
[19:43:23]: --- iPhone 6s - en-US ---
[19:43:23]: -------------------------
[19:43:23]: $ set -o pipefail && xcodebuild -workspace './project.xcworkspace' -scheme 'projectUITests' -derivedDataPath '/var/folders/00/y2fw4zqs2tqf84_w5skf4dhh0000gn/T/snapshot_derived20160519-13537-qw722a' -destination 'platform=iOS Simulator,id=93502B8C-D30E-46AE-9016-CDCFE0A7CA31,OS=9.3' clean build test | tee /Users/dev/Library/Logs/snapshot/project-projectUITests.log | xcpretty
```

**Comment by TKBurner** [comment]
Hi @esetnik Thanks for following up. Can you run this command and let me know if it results in an error?

Thanks!

`set -o pipefail && xcodebuild -workspace './project.xcworkspace' -scheme 'projectUITests' -derivedDataPath '/var/folders/00/y2fw4zqs2tqf84_w5skf4dhh0000gn/T/snapshot_derived20160519-13537-qw722a' -destination 'platform=iOS Simulator,id=93502B8C-D30E-46AE-9016-CDCFE0A7CA31,OS=9.3' clean build test | tee /Users/dev/Library/Logs/snapshot/project-projectUITests.log | xcpretty`

**Comment by esetnik** [comment]
Yes it results in an error after the simulator launches.
```
▸ Preprocessing /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Xcode/Agents/XCTRunner.app/Info.plist
▸ Running script '📦 Check Pods Manifest.lock'
▸ Compiling ProjectUITests.swift
▸ Compiling SnapshotHelper.swift
▸ Linking projectUITests
▸ Running script '📦 Copy Pods Resources'
▸ Touching projectUITests.xctest
** TEST FAILED **
```

**Comment by esetnik** [comment]
Hi @TKBurner how should I remove the `awaiting-reply` label?

**Comment by TKBurner** [comment]
Hi @esetnik Thanks for following up. Can you let me know if this issue persists if you set ios_version to 9.2? Also, can you try setting up Snapshot as it's own test target? Thanks!

**Comment by esetnik** [comment]
I get a different error when I try to use `ios_version "9.2"`

```
✗ snapshot
[11:23:02]: xcrun xcodebuild -list -workspace './campusSIMs.xcworkspace'

+----------------------------+------------------------------------------------------------------+
|                                  Summary for snapshot 1.12.2                                  |
+----------------------------+------------------------------------------------------------------+
| devices                    | ["iPhone 6s"]                                                    |
| languages                  | ["en-US"]                                                        |
| scheme                     | campusSIMsUITests                                                |
| clear_previous_screenshots | true                                                             |
| clean                      | true                                                             |
| workspace                  | ./campusSIMs.xcworkspace                                         |
| ios_version                | 9.2                                                              |
| output_directory           | /Users/ethan/work/campussims/campussims_ios/fastlane/screenshots |
| launch_arguments           | [""]                                                             |
| skip_open_summary          | false                                                            |
| reinstall_app              | false                                                            |
| erase_simulator            | false                                                            |
| localize_simulator         | false                                                            |
| app_identifier             | com.campusSIMs.campusSIMs                                        |
| buildlog_path              | ~/Library/Logs/snapshot                                          |
| number_of_retries          | 1                                                                |
| stop_after_first_error     | false                                                            |
+----------------------------+------------------------------------------------------------------+

[11:23:03]: Clearing previously generated screenshots
[11:23:03]: Building and running project - this might take some time...
[11:23:03]: Patching '/Users/ethan/Library/Preferences/com.apple.iphonesimulator.plist' to scale simulator to 100%
[11:23:05]: -------------------------
[11:23:05]: --- iPhone 6s - en-US ---
[11:23:05]: -------------------------
[11:23:05]: $ set -o pipefail && xcodebuild -workspace './campusSIMs.xcworkspace' -scheme 'campusSIMsUITests' -derivedDataPath '/var/folders/00/y2fw4zqs2tqf84_w5skf4dhh0000gn/T/snapshot_derived20160524-59195-ofe2ry' -destination 'platform=iOS Simulator,id=,OS=9.2' clean build test | tee /Users/ethan/Library/Logs/snapshot/campusSIMs-campusSIMsUITests.log | xcpretty
[11:23:05]: ▸ Loading...
[11:23:05]: ▸ xcodebuild: error: missing value for key 'id' of option 'Destination'
[11:23:05]: ▸ Usage: xcodebuild [-project <projectname>] [[-target <targetname>]...|-alltargets] [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
[11:23:05]: ▸ xcodebuild [-project <projectname>] -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
[11:23:05]: ▸ xcodebuild -workspace <workspacename> -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
[11:23:05]: ▸ xcodebuild -version [-sdk [<sdkfullpath>|<sdkname>] [<infoitem>] ]
[11:23:05]: ▸ xcodebuild -list [[-project <projectname>]|[-workspace <workspacename>]]
[11:23:05]: ▸ xcodebuild -showsdks
[11:23:05]: ▸ xcodebuild -exportArchive -archivePath <xcarchivepath> -exportPath <destinationpath> -exportOptionsPlist <plistpath>
[11:23:05]: ▸ xcodebuild -exportLocalizations -localizationPath <path> -project <projectname> [-exportLanguage <targetlanguage>...]
[11:23:05]: ▸ xcodebuild -importLocalizations -localizationPath <path> -project <projectname>
[11:23:05]: ▸ Options:
[11:23:05]: ▸ -usage                              print brief usage
[11:23:05]: ▸ -help                               print complete usage
[11:23:05]: ▸ -verbose                            provide additional status output
[11:23:05]: ▸ -license                            show the Xcode and SDK license agreements
[11:23:05]: ▸ -checkFirstLaunchStatus             Check if any First Launch tasks need to be performed
[11:23:05]: ▸ -project NAME                       build the project NAME
[11:23:05]: ▸ -target NAME                        build the target NAME
[11:23:05]: ▸ -alltargets                         build all targets
[11:23:05]: ▸ -workspace NAME                     build the workspace NAME
[11:23:05]: ▸ -scheme NAME                        build the scheme NAME
[11:23:05]: ▸ -configuration NAME                 use the build configuration NAME for building each target
[11:23:05]: ▸ -xcconfig PATH                      apply the build settings defined in the file at PATH as overrides
[11:23:05]: ▸ -arch ARCH                          build each target for the architecture ARCH; this will override architectures defined in the project
[11:23:05]: ▸ -sdk SDK                            use SDK as the name or path of the base SDK when building the project
[11:23:05]: ▸ -toolchain NAME                     use the toolchain with identifier or name NAME
[11:23:05]: ▸ -destination DESTINATIONSPECIFIER   use the destination described by DESTINATIONSPECIFIER (a comma-separated set of key=value pairs describing the destination to use)
[11:23:05]: ▸ -destination-timeout TIMEOUT        wait for TIMEOUT seconds while searching for the destination device
[11:23:05]: ▸ -parallelizeTargets                 build independent targets in parallel
[11:23:05]: ▸ -jobs NUMBER                        specify the maximum number of concurrent build operations
[11:23:05]: ▸ -dry-run                            do everything except actually running the commands
[11:23:05]: ▸ -hideShellScriptEnvironment         don't show shell script environment variables in build log
[11:23:05]: ▸ -showsdks                           display a compact list of the installed SDKs
[11:23:05]: ▸ -showBuildSettings                  display a list of build settings and values
[11:23:05]: ▸ -list                               lists the targets and configurations in a project, or the schemes in a workspace
[11:23:05]: ▸ -find-executable NAME               display the full path to executable NAME in the provided SDK and toolchain
[11:23:05]: ▸ -find-library NAME                  display the full path to library NAME in the provided SDK and toolchain
[11:23:05]: ▸ -version                            display the version of Xcode; with -sdk will display info about one or all installed SDKs
[11:23:05]: ▸ -enableAddressSanitizer YES|NO      turn the address sanitizer on or off when testing
[11:23:05]: ▸ -resultBundlePath PATH              specifies the directory where a result bundle describing what occurred will be placed
[11:23:05]: ▸ -derivedDataPath PATH               specifies the directory where build products and other derived data will go
[11:23:05]: ▸ -archivePath PATH                   specifies the directory where any created archives will be placed, or the archive that should be exported
[11:23:05]: ▸ -exportArchive                      specifies that an archive should be exported
[11:23:05]: ▸ -exportOptionsPlist PATH            specifies a path to a plist file that configures archive exporting
[11:23:05]: ▸ -enableCodeCoverage YES|NO          turn code coverage on or off when testing
[11:23:05]: ▸ -exportPath PATH                    specifies the destination for the product exported from an archive
[11:23:05]: ▸ -skipUnavailableActions             specifies that scheme actions that cannot be performed should be skipped instead of causing a failure
[11:23:05]: ▸ -exportLocalizations                exports completed and outstanding project localizations
[11:23:05]: ▸ -importLocalizations                imports localizations for a project, assuming any necessary localized resources have been created in Xcode
[11:23:05]: ▸ -localizationPath                   specifies a path to XLIFF localization files
[11:23:05]: ▸ -exportLanguage                     specifies multiple optional ISO 639-1 languages included in a localization export
xcodebuild: error: missing value for key 'id' of option 'Destination'

Usage: xcodebuild [-project <projectname>] [[-target <targetname>]...|-alltargets] [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
xcodebuild [-project <projectname>] -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
xcodebuild -workspace <workspacename> -scheme <schemeName> [-destination <destinationspecifier>]... [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...
xcodebuild -version [-sdk [<sdkfullpath>|<sdkname>] [<infoitem>] ]
xcodebuild -list [[-project <projectname>]|[-workspace <workspacename>]]
xcodebuild -showsdks
xcodebuild -exportArchive -archivePath <xcarchivepath> -exportPath <destinationpath> -exportOptionsPlist <plistpath>
xcodebuild -exportLocalizations -localizationPath <path> -project <projectname> [-exportLanguage <targetlanguage>...]
xcodebuild -importLocalizations -localizationPath <path> -project <projectname>

Options:
-usage                              print brief usage
-help                               print complete usage
-verbose                            provide additional status output
-license                            show the Xcode and SDK license agreements
-checkFirstLaunchStatus             Check if any First Launch tasks need to be performed
-project NAME                       build the project NAME
-target NAME                        build the target NAME
-alltargets                         build all targets
-workspace NAME                     build the workspace NAME
-scheme NAME                        build the scheme NAME
-configuration NAME                 use the build configuration NAME for building each target
-xcconfig PATH                      apply the build settings defined in the file at PATH as overrides
-arch ARCH                          build each target for the architecture ARCH; this will override architectures defined in the project
-sdk SDK                            use SDK as the name or path of the base SDK when building the project
-toolchain NAME                     use the toolchain with identifier or name NAME
-destination DESTINATIONSPECIFIER   use the destination described by DESTINATIONSPECIFIER (a comma-separated set of key=value pairs describing the destination to use)
-destination-timeout TIMEOUT        wait for TIMEOUT seconds while searching for the destination device
-parallelizeTargets                 build independent targets in parallel
-jobs NUMBER                        specify the maximum number of concurrent build operations
-dry-run                            do everything except actually running the commands
-hideShellScriptEnvironment         don't show shell script environment variables in build log
-showsdks                           display a compact list of the installed SDKs
-showBuildSettings                  display a list of build settings and values
-list                               lists the targets and configurations in a project, or the schemes in a workspace
-find-executable NAME               display the full path to executable NAME in the provided SDK and toolchain
-find-library NAME                  display the full path to library NAME in the provided SDK and toolchain
-version                            display the version of Xcode; with -sdk will display info about one or all installed SDKs
-enableAddressSanitizer YES|NO      turn the address sanitizer on or off when testing
-resultBundlePath PATH              specifies the directory where a result bundle describing what occurred will be placed
-derivedDataPath PATH               specifies the directory where build products and other derived data will go
-archivePath PATH                   specifies the directory where any created archives will be placed, or the archive that should be exported
-exportArchive                      specifies that an archive should be exported
-exportOptionsPlist PATH            specifies a path to a plist file that configures archive exporting
-enableCodeCoverage YES|NO          turn code coverage on or off when testing
-exportPath PATH                    specifies the destination for the product exported from an archive
-skipUnavailableActions             specifies that scheme actions that cannot be performed should be skipped instead of causing a failure
-exportLocalizations                exports completed and outstanding project localizations
-importLocalizations                imports localizations for a project, assuming any necessary localized resources have been created in Xcode
-localizationPath                   specifies a path to XLIFF localization files
-exportLanguage                     specifies multiple optional ISO 639-1 languages included in a localization export
[11:23:06]: Exit status: 64
[11:23:06]: Caught error... 64
```

> Also, can you try setting up Snapshot as it's own test target? 
I'm not exactly sure what I need to do for this. I have a scheme defined for UI tests and it only has the ui test target. I'm using that scheme in my `Snapfile`

```
# Uncomment the lines below you want to change by removing the # in the beginning

# A list of devices you want to take the screenshots from
devices([
   "iPhone 6s",
#   "iPhone 6 Plus",
#   "iPhone 5",
#   "iPhone 4s"
#   "iPad Retina",
#   "iPad Pro"
])

languages([
  "en-US",
#  "de-DE",
#  "it-IT",
#  ["pt", "pt_BR"] # Portuguese with Brazilian locale
])

# Arguments to pass to the app on launch. See https://github.com/fastlane/snapshot#launch-arguments
# launch_arguments(["-favColor red"])

# The name of the scheme which contains the UI Tests
scheme "campusSIMsUITests"

# Where should the resulting screenshots be stored?
# output_directory "./screenshots"

clear_previous_screenshots true

clean true

# Choose which project/workspace to use
# project "./Project.xcodeproj"
workspace "./campusSIMs.xcworkspace"

ios_version "9.2"
```

**Comment by TKBurner** [comment]
@esetnik Thanks for following up. Can you create a new test scheme and put your Snapshot implementation in there? 

Also can you set these manually:
derived_data_path and buildlog_path?

Thanks! 👍

**Comment by esetnik** [comment]
I found some interesting details when I inspected the manually set `buildlog_path`. This immediately proceeds the `snapshot` failure.

```
2016-05-25 17:09:15.581 XCTRunner[9814:8537433] Running tests...
2016-05-25 17:09:15.585 XCTRunner[9814:8537433] The bundle “campusSIMsUITests” couldn’t be loaded because it is damaged or missing necessary resources. Try reinstalling the bundle.
2016-05-25 17:09:15.585 XCTRunner[9814:8537433] (dlopen_preflight(/Users/ethan/work/campussims/campussims_ios/derived_data/Build/Products/Debug-iphonesimulator/campusSIMsUITests-Runner.app/PlugIns/campusSIMsUITests.xctest/campusSIMsUITests): Library not loaded: @rpath/Nimble.framework/Nimble
  Referenced from: /Users/ethan/work/campussims/campussims_ios/derived_data/Build/Products/Debug-iphonesimulator/campusSIMsUITests-Runner.app/PlugIns/campusSIMsUITests.xctest/campusSIMsUITests
  Reason: image not found)
```

**Comment by esetnik** [comment]
I removed nimble as a UI Test dependency and now I am generating screenshots successfully! I'm not sure why the tests pass when run from Xcode but fail from `snapshot` however I guess this has fixed the issue.

**Comment by TKBurner** [comment]
@esetnik Awesome! Really glad to hear you got up and running. Because we use a build tool we are more sensitive to build errors than Xcode itself. Thanks!

**Comment by minhnh94** [comment]
meet exactly same issue and I dont use any testing framework.
`snapshot` generates screenshots on iphone 5 with no problem but it always fails on iphone 6/6+.
Searched through hundreds of issues on github but no help :(

**Comment by TKBurner** [comment]
Hi @minhnh94 Can you please create a new issue with your full input and output? I'd love to take a look. Thanks so much!
